rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/shared_functions.hpp

```diff
@@ -229,6 +229,13 @@ wait(const char * implementation_identifier,
           }
         }
         if (fixed) {
+          // Reset the fixed guard conditions to avoid being woken up
+          // immediately next time.
+          DDSGuardCondition * guard = static_cast<DDSGuardCondition *>((*attached_conditions)[i]);
+          retcode = guard->set_trigger_value(DDS_BOOLEAN_FALSE);
+          if (retcode != DDS_RETCODE_OK) {
+            RMW_SET_ERROR_MSG("failed to set trigger value");
```

Comment:
OK, I made that change, and while I was at it, moved the reset logic to happen slightly earlier, to avoid having to recast the guard condition pointer later (no change in behavior). I accidentally squashed that last commit before pushing; sorry to do that while it's under review.